src/main/java/com/mparticle/kits/PilgrimSDKKit.java

```diff
@@ -0,0 +1,195 @@
+package com.mparticle.kits;
+
+import android.content.Context;
+
+import com.foursquare.pilgrim.LogLevel;
+import com.foursquare.pilgrim.PilgrimSdk;
+import com.foursquare.pilgrim.PilgrimUserInfo;
+import com.mparticle.MParticle;
+import com.mparticle.consent.ConsentState;
+import com.mparticle.identity.MParticleUser;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is the PilgrimSDK mParticle kit , used to extend the functionality of mParticle SDK and allow control of
+ * the PilgimSDK; Mapping analogous public mParticle APIs into PilgrimSDK's API.
+ * <p>
+ * <p>
+ * In addition to this file, you also will need to edit:
+ * - ./build.gradle (as explained above)
+ * - ./README.md
+ */
+final public class PilgrimSDKKit extends KitIntegration implements KitIntegration.UserAttributeListener, KitIntegration.IdentityListener {
+
+    /**
+     * Name of the kit
+     */
+    private static final String KIT_NAME = "PilgrimSDKKit";
+
+    /**
+     * Key used to get the api key from mParticle's settings response
+     */
+    private static final String PILGRIM_SDK_KEY = "pilgrim_sdk_key";
+
+    /**
+     * Key used to get the secret from mParticle's settings response
+     */
+    private static final String PILGRIM_SDK_SECRET = "pilgrim_sdk_secret";
+
+
+    private static final String ALLOW_DEBUG_OUTPUT = "enableDebug";
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        // We need to try and keep the context reference
+        // for us to try and stop the SDK when requested.
+        String key = settings.get(PILGRIM_SDK_KEY);
+        if (KitUtils.isEmpty(key)) {
+            throw new IllegalArgumentException("PilgrimSDK key is empty.");
+        }
+
+        String secret = settings.get(PILGRIM_SDK_SECRET);
+        if (KitUtils.isEmpty(secret)) {
+            throw new IllegalArgumentException("PilgrimSDK secret is empty.");
+        }
+
+        PilgrimSdk.Builder builder = new PilgrimSdk.Builder(context)
+                .consumer(key, secret)
+                .logLevel(LogLevel.ERROR);
+
+        if (KitUtils.parseBooleanSetting(settings, ALLOW_DEBUG_OUTPUT, false)) {
+            builder.enableDebugLogs();
+        }
+
+        // Configure with our starter
+        PilgrimSdk.with(builder);
+        // ----
+        PilgrimSdk.start(context);
```

Comment:
Don't think the kit should run `start`. That is usually done in an activity and we also have functionality that works without the SDK being started (ex: getting current location)

Reply:
Oh ok, wasn't sure if we had to start it or not for it to be configured.
I'll remove the .start part and let the user decide when to call it.